@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -550,13 +550,13 @@ export class TileGeometryCreator {
                 );
                 textElement.pathLengthSqr = textPath.pathLengthSqr;
                 textElement.minZoomLevel =
-                    technique.minZoomLevel !== undefined
-                        ? technique.minZoomLevel
-                        : mapView.minZoomLevel;
+                    getPropertyValue(technique.minZoomLevel, discreteZoomEnv) ??
+                    mapView.minZoomLevel;
+
                 textElement.maxZoomLevel =
-                    technique.maxZoomLevel !== undefined
-                        ? technique.maxZoomLevel
-                        : mapView.maxZoomLevel;
+                    getPropertyValue(technique.maxZoomLevel, discreteZoomEnv) ??
+                    mapView.maxZoomLevel;
```

Comment:
you can remove the parts that default to mapView.min/maxZoomLevel, the limits can be left undefined.